packages/models-library/tests/test_user_preferences.py

```diff
@@ -0,0 +1,162 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+from collections.abc import Iterator
+from copy import deepcopy
+from typing import Any
+
+import arrow
+import pytest
+from models_library.services import ServiceKey
+from models_library.user_preferences import (
+    BaseBackendUserPreference,
+    BaseFrontendUserPreference,
+    BaseUserServiceUserPreference,
+    PreferenceType,
+    _AutoRegisterMeta,
+    _BaseUserPreferenceModel,
+)
+from pydantic import parse_obj_as
+
+_SERVICE_KEY_SAMPLES: list[str] = [
+    "simcore/services/comp/something-1231",
+    "simcore/services/dynamic/something-1231",
+    "simcore/services/frontend/something-1231",
+]
+
+
+@pytest.fixture(params=[None, 1, 1.0, "str", {"a": "dict"}, ["a", "list"]])
+def value(request: pytest.FixtureRequest) -> Any:
+    return request.param
+
+
+def _get_base_user_preferences_data(
+    preference_type: PreferenceType, value: Any
+) -> dict[str, Any]:
+    return {"preference_type": preference_type, "value": value}
+
+
+def _get_utc_timestamp() -> float:
+    return arrow.utcnow().datetime.timestamp()
+
+
+@pytest.mark.parametrize("preference_type", PreferenceType)
+def test_base_user_preference_model(value: Any, preference_type: PreferenceType):
+    base_data = _get_base_user_preferences_data(
+        preference_type=preference_type, value=value
+    )
+    assert parse_obj_as(_BaseUserPreferenceModel, base_data)
+
+
+def test_backend_preferences(value: Any):
+    base_data = _get_base_user_preferences_data(
+        preference_type=PreferenceType.BACKEND, value=value
+    )
+    backend_preference = parse_obj_as(BaseBackendUserPreference, base_data)
+    # check serialization
+    assert set(backend_preference.dict().keys()) == {"value"}
+
+
+def test_frontend_preferences(value: Any):
+    base_data = _get_base_user_preferences_data(
+        preference_type=PreferenceType.FRONTEND, value=value
+    )
+
+    data_with_rendered_widget = deepcopy(base_data)
+    data_with_rendered_widget.update(
+        {
+            "preference_identifier": "pref-name",
```

Comment:
I do not quite get the idea with this preference identifier. why not storing it in the db if the frontend then retrieves it?
I still do not see the added value of not storing it. same with that last_modified_timestamp_utc variable actually.